test/SILGen/statements.swift

```diff
@@ -657,6 +657,113 @@ func test_as_pattern(_ y : BaseClass) -> DerivedClass {
+  // expected-warning@-1 {{immutable value 't' was never used; consider replacing with '_' or removing it}}
+  // CHECK: [[SET_CAST_FN:%[0-9]+]] = function_ref @$ss23_setDownCastConditionalyShyq_GSgShyxGSHRzSHR_r0_lF : $@convention(thin) <τ_0_0, τ_0_1 where τ_0_0 : Hashable, τ_0_1 : Hashable> (@guaranteed Set<τ_0_0>) -> @owned Optional<Set<τ_0_1>>
+  // CHECK-NEXT: [[RESULT:%[0-9]+]] = apply [[SET_CAST_FN]]<T, Bool>
+  // CHECK-NEXT: switch_enum [[RESULT]]
+  guard case let t as Set<Bool> = set else {}
+  // expected-warning@-1 {{immutable value 't' was never used; consider replacing with '_' or removing it}}
+  // CHECK: [[SET_CAST_FN:%[0-9]+]] = function_ref @$ss23_setDownCastConditionalyShyq_GSgShyxGSHRzSHR_r0_lF : $@convention(thin) <τ_0_0, τ_0_1 where τ_0_0 : Hashable, τ_0_1 : Hashable> (@guaranteed Set<τ_0_0>) -> @owned Optional<Set<τ_0_1>>
+  // CHECK-NEXT: [[RESULT:%[0-9]+]] = apply [[SET_CAST_FN]]<T, Bool>
+  // CHECK-NEXT: switch_enum [[RESULT]]
+  while case let t as Set<Bool> = set {}
+  // expected-warning@-1 {{immutable value 't' was never used; consider replacing with '_' or removing it}}
+}
+// CHECK: } // end sil function '$s10statements28test_as_pattern_set_downcastyyShyxGSHRzlF'
+
+// CHECK-LABEL: sil hidden [ossa] @$s10statements35test_is_pattern_collection_downcastyySayAA9BaseClassCGF : $@convention(thin) (@guaranteed Array<BaseClass>) -> () {
+func test_is_pattern_collection_downcast(_ arr: [BaseClass]) {
+  // CHECK: [[ARR_CAST_FN:%[0-9]+]] = function_ref @$ss21_arrayConditionalCastySayq_GSgSayxGr0_lF : $@convention(thin) <τ_0_0, τ_0_1> (@guaranteed Array<τ_0_0>) -> @owned Optional<Array<τ_0_1>>
+  // CHECK-NEXT: [[RESULT:%[0-9]+]] = apply [[ARR_CAST_FN]]<BaseClass, DerivedClass>
+  // CHECK-NEXT: switch_enum [[RESULT]]
+  if case is [DerivedClass] = arr {}
+  // CHECK: [[ARR_CAST_FN:%[0-9]+]] = function_ref @$ss21_arrayConditionalCastySayq_GSgSayxGr0_lF : $@convention(thin) <τ_0_0, τ_0_1> (@guaranteed Array<τ_0_0>) -> @owned Optional<Array<τ_0_1>>
+  // CHECK-NEXT: [[RESULT:%[0-9]+]] = apply [[ARR_CAST_FN]]<BaseClass, DerivedClass>
+  // CHECK-NEXT: switch_enum [[RESULT]]
+  guard case is [DerivedClass] = arr else {}
+  // CHECK: [[ARR_CAST_FN:%[0-9]+]] = function_ref @$ss21_arrayConditionalCastySayq_GSgSayxGr0_lF : $@convention(thin) <τ_0_0, τ_0_1> (@guaranteed Array<τ_0_0>) -> @owned Optional<Array<τ_0_1>>
+  // CHECK-NEXT: [[RESULT:%[0-9]+]] = apply [[ARR_CAST_FN]]<BaseClass, DerivedClass>
+  // CHECK-NEXT: switch_enum [[RESULT]]
+  while case is [DerivedClass] = arr {}
+}
+// CHECK: } // end sil function '$s10statements35test_is_pattern_collection_downcastyySayAA9BaseClassCGF'
+
+// CHECK-LABEL: sil hidden [ossa] @$s10statements35test_is_pattern_dictionary_downcastyySDySSAA9BaseClassCGF : $@convention(thin) (@guaranteed Dictionary<String, BaseClass>) -> () {
+func test_is_pattern_dictionary_downcast(_ dict: Dictionary<String, BaseClass>) {
```

Comment:
All these tests are really similar so I suggest we group them by collection kind 3 functions named `test_isa_pattern_*_downcast`.

Reply:
Seems like you resolved the conversation, but forgot to address the suggestion?

Reply:
fixed that!

Reply:
I did change the name of these functions. Did you mean something else?

Reply:
Ah, right, what I meant is to group these tests by collection kind into 3 functions rather than 6. Each function then tests both `is` and `as` patterns for a given collection kind.